cub/cub/device/dispatch/kernels/warpspeed/optimizeSmemPtr.cuh

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+#pragma once
+
+#include <cub/config.cuh>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+CUB_NAMESPACE_BEGIN
+
+template <typename _Tp>
+[[nodiscard]] _CCCL_DEVICE_API inline _Tp* optimizeSmemPtr(const _Tp* smemGeneric)
```

Comment:
Question: This is coming up more often that we like. Should we move this internal helper into libcu++ and make it `__ugly`